superset/charts/api.py

```diff
@@ -591,25 +603,36 @@ def cache_screenshot(self, pk: int, **kwargs: Any) -> WerkzeugResponse:
 
         chart_url = get_url_path("Superset.slice", slice_id=chart.id)
         screenshot_obj = ChartScreenshot(chart_url, chart.digest)
-        cache_key = screenshot_obj.cache_key(window_size, thumb_size)
+        cache_key = screenshot_obj.get_cache_key(window_size, thumb_size)
+        cache_payload = (
+            screenshot_obj.get_from_cache_key(cache_key) or ScreenshotCachePayload()
+        )
         image_url = get_url_path(
             "ChartRestApi.screenshot", pk=chart.id, digest=cache_key
         )
 
-        def trigger_celery() -> WerkzeugResponse:
+        def build_response(status_code: int) -> WerkzeugResponse:
+            return self.response(
+                status_code,
+                cache_key=cache_key,
+                chart_url=chart_url,
+                image_url=image_url,
+                updated_at=cache_payload.get_timestamp(),
+                update_status=cache_payload.get_status(),
+            )
+
+        if cache_payload.status in [StatusValues.PENDING, StatusValues.ERROR] or force:
```

Comment:
Typically when a caching operation has errored out, we'd actually like to leave it in the cache with a specific TTL, as there's usually some access issue, or the chart is broken or something similar. Rerunning that screenshot every time is very expensive. Could we maybe introduce a config flag with an error TTL? This will likely save considerable compute on large deployments.